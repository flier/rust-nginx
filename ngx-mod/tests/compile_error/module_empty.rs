use ngx_mod::{rt::ffi, Module};

#[derive(Module)]
struct M;

#[no_mangle]
static mut ngx_m_module_commands: [ffi::ngx_command_t; 0] = [];

fn main() {}
