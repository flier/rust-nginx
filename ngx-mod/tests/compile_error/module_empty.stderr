error[E0277]: the trait bound `M: ngx_mod::Module` is not satisfied
 --> tests/compile_error/module_empty.rs
  |
  | #[derive(Module)]
  |          ^^^^^^ the trait `ngx_mod::Module` is not implemented for `M`
  |
  = note: required for `M` to implement `ngx_mod::UnsafeModule`
  = note: this error originates in the derive macro `Module` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `M: ngx_mod::http::Module` is not satisfied
 --> tests/compile_error/module_empty.rs
  |
  | #[derive(Module)]
  |          ^^^^^^ the trait `ngx_mod::http::Module` is not implemented for `M`
  |
  = note: required for `M` to implement `ngx_mod::http::UnsafeModule`
  = note: this error originates in the derive macro `Module` (in Nightly builds, run with -Z macro-backtrace for more info)
